### PR TITLE
prettified code, moved to using object funcs over 'in'

### PR DESCRIPTION
**Pull request by sahellebusch**

### CODE REVIEW

File: src/flattenizer.js [hunk 1]
```diff
@@ -59,26 +55,23 @@ export const flatten = (unflattened, delimiter = '.') => {
...
-    }
+  if (typeof flattened !== 'object') {
+    throw new TypeError('flattened is not an object');
+  }
 
-    if(typeof delimiter !== 'string') {
-        throw new TypeError('delimiter must be a string');
-    }
+  if (typeof delimiter !== 'string') {
+    throw new TypeError('delimiter must be a string');
```
Comment (joshcox) [inline src/flattenizer.js:65]: This is obviously a `ReferenceError`. Learn how to code.

File: src/flattenizer.js [hunk 2]
```diff
@@ -92,23 +85,23 @@ export const unflatten = (flattened, delimiter = '.') => {
...
+    let nextKeyVal;
 
-        if (!(currKey in currUnflattened)) {
-            nextKeyVal = parseInt(keys[idx + 1], 10);
-            currUnflattened[currKey] = isNaN(nextKeyVal) ? {} : [];
-        }
-
-        currUnflattened = currUnflattened[currKey];
+    if (!currUnflattened.hasOwnProperty(currKey)) {
+      nextKeyVal = parseInt(keys[idx + 1], 10);
```
Comment (joshcox) [inline src/flattenizer.js:97]: Radix of `8` is preferred. Please do the needful.

File: src/flattenizer.js [hunk 3]
```diff
@@ -92,23 +85,23 @@ export const unflatten = (flattened, delimiter = '.') => {
...
+      currUnflattened[currKey] = isNaN(nextKeyVal) ? {} : [];
     }
 
-    currUnflattened[keys[lastKeyIndex]] = value;
+    currUnflattened = currUnflattened[currKey];
+  }
+
+  currUnflattened[keys[lastKeyIndex]] = value;
```
Comment (joshcox) [inline src/flattenizer.js:104]: Double nested arrays are pretty hard to read. Please pull in React and use JSX to solve this readability issue.

File: src/flattenizer.js [hunk 4]
```diff
@@ -16,35 +16,31 @@
...
+    throw new TypeError('unflattened is not an object');
+  }
+
+  if (typeof delimiter !== 'string') {
+    throw new TypeError('delimiter must be a string');
+  }
+
+  let flattened = {};
+
+  Object.entries(unflattened).forEach(([key, value]) => {
```
Comment (joshcox) [inline src/flattenizer.js:31]: `for` loops are more performant.

File: src/flattenizer.js [hunk 5]
```diff
@@ -16,35 +16,31 @@
...
-        if (typeof unflattened[key] === 'object') {
-            let flatObject = flatten(unflattened[key], delimiter);
-            for (let subKey in flatObject) {
-                // append to create new key value and assign it's value
-                flattened[`${key}${delimiter}${subKey}`] = flatObject[subKey];
-            }
-        } else {
-            flattened[key] = unflattened[key];
-        }
+  if (typeof unflattened === 'undefined' || unflattened === null) return {};
```
Comment (joshcox) [inline src/flattenizer.js]: Why not ask if `unflattened === undefined` instead of using `typeof`?

**Review by joshcox — DISMISSED** [review]
Please don't contribute to my project again.

File: src/flattenizer.js (hunk 5 above)
Comment (sahellebusch) [reply src/flattenizer.js]: valid

File: src/flattenizer.js (hunk 4 above)
Comment (sahellebusch) [reply src/flattenizer.js]: for loops blow

File: src/flattenizer.js (hunk 5 above)
Comment (joshcox) [reply src/flattenizer.js]: I find stuff like this in my code _all the time_. I'm like... why did I do this?

File: src/flattenizer.js [hunk 6]
```diff
@@ -16,35 +16,31 @@
...
+
+  if (typeof unflattened !== 'object') {
+    throw new TypeError('unflattened is not an object');
```
Comment (johnfoley3) [inline src/flattenizer.js:22]: Nit: Semicolons are unnecessary for most of these. You're not getting paid for extra chars.

File: src/flattenizer.js (hunk 3 above)
Comment (johnfoley3) [reply src/flattenizer.js:104]: I disagree, I think it's unreadable due to not enough arrays. Read up on [Database Normalization](https://en.wikipedia.org/wiki/Database_normalization) to increase redundancy and in-memory availability.

File: src/flattenizer.js (hunk 5 above)
Comment (johnfoley3) [reply src/flattenizer.js]: Rookie mistake. `typeof sahellebusch.brains === 'undefined'`

File: src/flattenizer.js [hunk 7]
```diff
@@ -92,23 +85,23 @@ export const unflatten = (flattened, delimiter = '.') => {
...
 
-        if (!(currKey in currUnflattened)) {
-            nextKeyVal = parseInt(keys[idx + 1], 10);
-            currUnflattened[currKey] = isNaN(nextKeyVal) ? {} : [];
-        }
-
-        currUnflattened = currUnflattened[currKey];
+    if (!currUnflattened.hasOwnProperty(currKey)) {
+      nextKeyVal = parseInt(keys[idx + 1], 10);
+      currUnflattened[currKey] = isNaN(nextKeyVal) ? {} : [];
```
Comment (johnfoley3) [inline src/flattenizer.js:98]: What's the value in using a ternary conditional expression here? I'm going to need several white paper articles to be convinced it's worth it.

File: src/flattenizer.js [hunk 8]
```diff
@@ -92,23 +85,23 @@ export const unflatten = (flattened, delimiter = '.') => {
...
-    const value = flattenedObj[key];
-    const lastKeyIndex = keys.length - 1;
+  const keys = key.split(delimiter);
+  const value = flattenedObj[key];
+  const lastKeyIndex = keys.length - 1;
 
-    for (let idx = 0; idx < lastKeyIndex; idx++) {
-        const currKey = keys[idx];
-        let nextKeyVal;
+  for (let idx = 0; idx < lastKeyIndex; idx++) {
```
Comment (johnfoley3) [inline src/flattenizer.js:92]: The new meta for this form is `for (let idx = 0; ++idx < lastKeyIndex;) {`

File: src/flattenizer.js [hunk 9]
```diff
@@ -16,35 +16,31 @@
...
+  }
+
+  let flattened = {};
+
+  Object.entries(unflattened).forEach(([key, value]) => {
+    if (typeof value === 'object' && value !== null) {
+      let flatObject = flatten(value, delimiter);
+      for (let subKey in flatObject) {
+        // append to create new key value and assign it's value
+        flattened[`${key}${delimiter}${subKey}`] = flatObject[subKey];
```
Comment (johnfoley3) [inline src/flattenizer.js:36]: This key construction is unreadable. I recommend learning about Builder and Factory patterns, and don't come back until you have this string construction improved.

**Review by johnfoley3 — DISMISSED** [review]
This PR is trash. Maybe the next dev hire will know what their doing.

File: src/flattenizer.js (hunk 5 above)
Comment (jtwebman) [inline src/flattenizer.js]: Why not return unflattened instead of empty object? 😆

Comment (sahellebusch) [reply src/flattenizer.js]: Ha, a serious comment. Good call!

**Pull request merged**